app/scripts/models/auth_brokers/redirect.js

```diff
@@ -76,7 +76,7 @@ define([
       // event handlers before the flow completes.
       var self = this;
       return p().delay(100).then(function () {
-        if (self.isOriginalTab()) {
+        if (self.isOriginalTab() || account.get('sessionToken') && self.relier.get('alwaysRedirect')) {
```

Comment:
Comment copy from the previous revision:
![image](https://cloud.githubusercontent.com/assets/128755/7460694/01eeef38-f273-11e4-9a73-c70211f6a966.png)